src/MongoDB/Cursor.c

```diff
@@ -509,7 +509,7 @@ static HashTable *php_phongo_cursor_get_debug_info(zval *object, int *is_temp TS
 	if (!Z_ISUNDEF(intern->visitor_data.zchild)) {
 #if PHP_VERSION_ID >= 70000
 		ADD_ASSOC_ZVAL_EX(&retval, "currentDocument", &intern->visitor_data.zchild);
-		/*Z_ADDREF(intern->visitor_data.zchild);*/
+		Z_ADDREF(intern->visitor_data.zchild);
```

Comment:
Where are you removing this reference again? And why is this different for PHP 7+ and PHP 7- ?

Reply:
If you look at the original commit (https://github.com/mongodb/mongo-php-driver/commit/78839bde83d3df0d5775485a383d2e67e8a0e969#diff-d11812fede193d119daadf3b31af5049R344), I think this was just over oversight. This change makes PHP 7+ consistent with 5.x (and other zvals in the debug output on all PHP versions) by always adding a reference.

Is it necessary to free the reference in debug info? My understanding is that it's essentially a return value and the root array (or HashTable) will get freed by the caller.

Reply:
Is there a concern that we're only adding a pointer to the current document here instead of copying the zval and it could change later if the user were to hold a reference to it (not with `var_dump()` but presumably some other method).

I'll note that `php_phongo_cursor_free_current()` is going to `zval_ptr_dtor()` and `ZVAL_UNDEF()` at each step of iteration.